lib/common.py

```diff
@@ -65,8 +64,23 @@ def xo_object_exists(uuid):
     lst = json.loads(xo_cli('--list-objects', {'uuid': uuid}))
     return len(lst) > 0
 
-class SSHException(Exception):
-    pass
+class SSHCommandFailed(Exception):
+    __slots__ = 'returncode', 'stdout'
```

Comment:
not needed anymore since you define the attributes in `__init__`? Else `cmd` is missing.

Reply:
`__slots__` is useful to increase the performance when a property is accessed. I often use this attribute when the class is small. But yes, I must add `cmd` here. :)

Reply:
The docs I read suggest this performance improvement is mostly useful when there are going to be a lot of objects (thousands, millions), so now we're ready for some intensive testing!

Reply:
Yes, but the another benefice is the documentation. I don't like the dynamic declaration of attributes in this language. :)